core/src/test/scala/org/apache/spark/util/ThreadUtilsSuite.scala

```diff
@@ -66,4 +66,24 @@ class ThreadUtilsSuite extends SparkFunSuite {
     val futureThreadName = Await.result(f, 10.seconds)
     assert(futureThreadName === callerThreadName)
   }
+
+  test("runInNewThread") {
+    import ThreadUtils._
+    assert(runInNewThread("thread-name") { Thread.currentThread().getName } === "thread-name")
+    assert(runInNewThread("thread-name") { Thread.currentThread().isDaemon } === true)
+    assert(
+      runInNewThread("thread-name", isDaemon = false) { Thread.currentThread().isDaemon } === false
+    )
+    val exception = intercept[IllegalArgumentException] {
+      runInNewThread("thread-name") { throw new IllegalArgumentException("test") }
```

Comment:
can you add some kind of more unique identifier here? I wouldn't be surprised if elsewhere we have `test` in the stack trace. Maybe just do `"test-" + System.currentTimeMillis`